src/Gui/Tree.cpp

```diff
@@ -3061,9 +3066,17 @@ void TreeWidget::onUpdateStatus()
                 errors.push_back(obj);
             if (docItem->ObjectMap.count(obj))
                 continue;
-            auto vpd = Base::freecad_dynamic_cast<ViewProviderDocumentObject>(gdoc->getViewProvider(obj));
+            auto vpd =
+                Base::freecad_dynamic_cast<ViewProviderDocumentObject>(gdoc->getViewProvider(obj)); 

```

Comment:
`Base::freecad_dynamic_cast<T>()` has been replaced with `freecad_cast<T*>()`:
* https://github.com/FreeCAD/FreeCAD/pull/20515
* https://github.com/FreeCAD/FreeCAD/pull/20694

I think this might be the reason why this branch has a collision, fixing it would look like:

```suggestion
            auto vpd = freecad_cast<ViewProviderDocumentObject*>(gdoc->getViewProvider(obj)); 
```